doc/user/flowspec.rst

```diff
@@ -0,0 +1,330 @@
+.. _flowspec:
+
+BGP Flowspec
+============
+
+.. _features-of-the-current-implementation-flowspec:
+
+Flowspec overview
+------------------
+
+BGP flowspec introduces a new Network Layer Reachability Information (NLRI) encoding
+format that is used to distribute traffic rule flow specifications. Basically,
+instead of simply relying on destination IP address for IP prefixes, the IP prefix
+is replaced by a n-tuple consisting of a rule. That rule can be a more or less
+complex combination of the following:
+
+
+- Network source/destination (can be one or the other, or both).
+- Layer 4 information for UDP/TCP : source port, or destination port, or any port.
+- Layer 4 information for ICMP type and ICMP code.
+- Layer 4 information for TCP Flags
+- Layer 3 information : DSCP value, Protocol type, packet length, fragmentation.
+- Misc layer 4 TCP flags.
+
+A combination of the above rules is applied for traffic filtering. This is encoded
+as part of specific BGP extended communities and the action can range from the
+obvious rerouting (to nexthop or to separate VRF) to shaping, or discard.
+
+Following IETF drafts and RFCs have been used to implement FRR Flowspec:
+
+- [RFC5575]_
+- [Draft IETF IDR Flowspec redirect IP]_
+
+
+.. _design-principles-flowspec:
+
+
+Design Principles
+-----------------
+
+FRR implements the Flowspec client side, that is to say that BGP is able to
+receive Flowspec entries, but is not able to act as manager and send Flowspec entries.
+
+FRRouting uses policy based routing mechanisms to apply Flowspec configuration to the
+underlying system. The underlying system is Linux, and provides a rich enough API to
+implement this policy routing.
+
+- by filtering the traffic based on `Linux NetFilter`.
+
+`Linux NetFilter` provides a set of tools like ipset and iptables that are powerful
```

Comment:
not sure about what you mean by "Also, please indent this to match the indentation of the word by above. Otherwise this will be two lists with one item each instead of one list with two items."

here is what I did. is it ok?

```
Linux provides the following mechanisms to implement policy based routing:

- Filtering the	traffic	with NetFilter.
Netfilter provides a set of tools like ``ipset`` and ``iptables`` that are powerful
enough to be able to filter such Flowspec filter rule.

- using	non standard routing tables via	``iproute2`` (via the ``ip rule`` command provided
by ``iproute2``).
```